tensorboard/webapp/widgets/line_chart_v2/lib/scale.ts

```diff
@@ -148,28 +149,13 @@ class Log10Scale implements Scale {
 
     const adjustedMin = Math.max(min, Number.MIN_VALUE);
     const adjustedMax = Math.max(max, Number.MIN_VALUE);
-    if (min <= 0 || max <= 0) {
-      return [adjustedMin, adjustedMax];
+
+    if (max <= 0) {
+      // When both min and max are non-positive, clip to [Number.MIN_VALUE, 1].
+      return [Number.MIN_VALUE, 1];
     }
 
-    const numericMinLogValue = this.transform(Number.MIN_VALUE);
-    const minLogValue = this.transform(adjustedMin);
-    const maxLogValue = this.transform(adjustedMax);
-
-    const spreadInLog = maxLogValue - minLogValue;
-    const padInLog =
-      spreadInLog > 0
-        ? spreadInLog * PADDING_RATIO
-        : // In case `minLogValue` is 0 (i.e., log_10(1) = 0), we want some padding.
-          Math.max(
-            Math.abs(minLogValue * PADDING_RATIO),
-            MIN_SIGNIFICANT_PADDING
-          );
-
-    return [
-      this.untransform(Math.max(numericMinLogValue, minLogValue - padInLog)),
-      this.untransform(maxLogValue + padInLog),
-    ];
+    return [Math.max(Number.MIN_VALUE, adjustedMin * 0.5), adjustedMax * 1.5];
```

Comment:
You mentioned "Put some constant padding in the log scale (halving and doubling in log scale...)" but this is halving and then adding 50% - should it be `adjustedMax * 2.0`?  Or do we want the upper padding to be smaller for some reason?

This also seems like it doesn't match the tests which might be causing the failures (although weirdly, only for one of the two travis jobs).

Reply:
Yup, that was a mistake on my part. Fixed the test and code.